docs/excel/custom-functions-architecture.md

```diff
@@ -0,0 +1,50 @@
+---
+ms.date: 03/20/2019
+description: Learn about Excel custom functions' runtime. 
+title: Custom functions architecture (preview)
+localization_priority: Priority
+---
+# Custom functions architecture
+
+ Custom functions are with their own unique runtime that prioritizes execution of calculations. This article will cover the differences between the custom functions runtime and the browser-based JavaScript engine which powers most other parts of your add-in.
+
+## Custom functions runtime
+
+Because of their unique runtime, think of custom functions as another component of your total add-in. Your add-in will also likely include elements from the browser engine runtime, like a task pane.
+
+The following table highlights the differences between the custom functions runtime and the browser engine runtime:
+
+| Custom functions runtime 	| Browser engine runtime 	|
+|------------------------------------------------------------------	|--------------------------------------------------------------------------------------------------------------	|
+| Supports returning a value from a cell 	| Supports Office.js APIs and UI elements 	|
+| Does not have `localStorage` object, instead uses `AsyncStorage` 	| Has `localStorage` object, can optionally use `AsyncStorage` object 	|
+| Provides access to `OfficeRuntime` objects 	| Provides access to `OfficeRuntime` objects 	|
+| Allows you to make web requests 	| Allows you to access the Document Object Model (DOM) and supports libraries that use the DOM, such as jQuery 	|
+
+>[!NOTE]
+> When first installing a custom function, a task pane is shown. This pane is necessary to install the custom functions runtime, but shouldn't serve as the task pane for your add-in.
+
+## Browser engine runtime
+
+The task pane and other components that are not custom functions run in a browser engine runtime.
+
+The browser engine runtime enables access to the Office.js APIs. Keep in mind that any of the Excel APIs, such as APIs which allow you to manipulate Excel tables, run on the browser engine runtime, but aren't directly accessible from the custom functions runtime.
+
+## Communicate between runtimes
+
+Both the browser engine runtime and the custom functions runtime have access to `OfficeRuntime.AsyncStorage` and `OfficeRuntime.Dialog` APIs.
```

Comment:
Maybe start off with a scenario to give this more of a why.

suggestion...
"Code in your custom functions cannot directly interact with code in other parts of your web add-in, like the task pane because they are in different runtimes. But in some scenarios you may need to share data, such as passing a token. AsyncStorage can be used to store data..."